consensus/src/validation.rs

```diff
@@ -0,0 +1,483 @@
+use std::fmt::{Debug, Display, Formatter, Result as FmtResult};
+
+use crate::{
+    alerts::{Alert, ForkingNotification},
+    units::{
+        SignedUnit, UncheckedSignedUnit, Unit, UnitStore, UnitStoreStatus, ValidationError,
+        Validator as UnitValidator, WrappedUnit,
+    },
+    Data, Hasher, MultiKeychain, NodeIndex, NodeSubset, Round,
+};
+
+/// What can go wrong when validating a unit.
+#[derive(Eq, PartialEq)]
+pub enum Error<H: Hasher, D: Data, MK: MultiKeychain> {
+    Invalid(ValidationError<H, D, MK::Signature>),
+    Duplicate(SignedUnit<H, D, MK>),
+    Uncommitted(SignedUnit<H, D, MK>),
+    NewForker(Box<Alert<H, D, MK::Signature>>),
```

Comment:
Do we need a `Box` here?

Reply:
Hmm I think its strange to return alert in the error variant. I think more natural way to do it would be including just necessary extra information like other forker unit for example.

Reply:
In somewhat good news this becomes more natural in the next part, where handling of these errors gets factored out of runway. Although it remains an error variant internally, and returns the whole alert (it would be somewhat annoying to do this differently, since we only have all the necessary info inside of `validation`).